esp/src/src-react/components/HexView.tsx

```diff
@@ -0,0 +1,175 @@
+import * as React from "react";
+import { Checkbox, ICheckboxStyles, ISpinButtonStyles, Label, SpinButton } from "@fluentui/react";
+import { Topology, Workunit } from "@hpcc-js/comms";
+import { TextSourceEditor } from "./SourceEditor";
+import nlsHPCC from "src/nlsHPCC";
+import { scopedLogger } from "@hpcc-js/util";
+
+const logger = scopedLogger("src-react/components/HexView.tsx");
+
+interface HexViewProps {
+    logicalFile: string;
+}
+
+const bufferLength = 16 * 1024;
+const unknownChar = String.fromCharCode(8226);
+
+const spinStyles: Partial<ISpinButtonStyles> = { root: { width: 100 }, spinButtonWrapper: { width: 50 } };
+const checkboxStyles: Partial<ICheckboxStyles> = { root: { marginTop: 4 } };
+
+const isCharPrintable = (char) => {
+    const charCode = char.charCodeAt(0);
+    if (charCode < 32) return false;
+    else if (charCode >= 127 && charCode <= 159) return false;
+    else if (charCode === 173) return false;
+    else if (charCode > 255) return false;
+    return true;
+};
+
+export const HexView: React.FunctionComponent<HexViewProps> = ({
+    logicalFile
+}) => {
+
+    const [text, setText] = React.useState("");
+    const [cachedResponse, setCachedResponse] = React.useState<any[]>([]);
+    const [lineLength, setLineLength] = React.useState("16");
+    const [showEbcdic, setShowEbcdic] = React.useState(false);
+    const [queryTarget, setQueryTarget] = React.useState("");
+
+    const onLineLengthChange = React.useCallback((event: React.FormEvent<HTMLInputElement>, newValue?: string) => {
+        if (newValue !== undefined) {
+            setLineLength(newValue);
+        }
+    }, []);
+
+    const query = React.useMemo(() => {
+        return `data_layout := record
+    data1 char;
+end;
+data_dataset := dataset('${logicalFile}', data_layout, thor);
+analysis_layout := record
+    data1 char;
+    string1 str1;
+    ebcdic string1 estr1;
+end;
+analysis_layout calcAnalysis(data_layout l) := transform
+    self.char := l.char;
+    self.str1 := transfer(l.char, string1);
+    self.estr1 := transfer(l.char, string1);
+end;
+analysis_dataset := project(data_dataset, calcAnalysis(left));
+choosen(analysis_dataset, ${bufferLength});`;
+    }, [logicalFile]);
+
+    React.useEffect(() => {
+        const topology = Topology.attach({ baseUrl: "" });
+        topology.fetchLogicalClusters().then(response => {
+            setQueryTarget(response.filter(row => row.Type === "hthor")[0]?.Name || "");
+        }).catch(err => logger.error(err));
```

Comment:
I do have a slightly different heuristic elsewhere - will common up with this once its pulled (quicker than trying to articulate it here).